docs/docs/guides/next.mdx

```diff
@@ -6,72 +6,113 @@ sidebar_custom_props:
 
-## Install dependencies
+Learn how to quickly add authentication and user profile in your Next.js app using Hanko.
 
-Install the `@teamhanko/hanko-elements` package:
+## Install @teamhanko/hanko-elements
+
+Once you've initialized your Next app, installing hanko-elements provides you with access to the prebuilt components: `hanko-auth` and `hanko-profile`.
 
 ```shell npm2yarn
 npm install @teamhanko/hanko-elements
 ```
 
+## Add the Hanko API URL
+
+Retrieve the API URL from the [Hanko console](https://cloud.hanko.io/) and place it in your .env file.
+
+```sh title=".env"
+HANKO_API_URL=https://f4****-4802-49ad-8e0b-3d3****ab32.hanko.io
+```
+
 ## Add `<hanko-auth>` component
 
-To provide a login interface in your app, use the `<hanko-auth>` web component. To do so, first import the `register` function
-from `@teamhanko/hanko-elements` in your Next.js component. Call it with the URL of the Hanko API as an argument to register
-the `<hanko-auth>` element with
-the browser's [`CustomElementRegistry`](https://developer.mozilla.org/de/docs/Web/API/CustomElementRegistry).
-Then use the element in your JSX.
+The `<hanko-auth>` web component adds a login interface to your app. Begin by importing the register function from `@teamhanko/hanko-elements` into your Next.js component. Call it with the Hanko API URL as an argument to register `<hanko-auth>` with the browser's [CustomElementRegistry](https://developer.mozilla.org/de/docs/Web/API/CustomElementRegistry). Once done, include it in your JSX.
 
-:::info
+```mdx-code-block
+<Tabs groupId="environment">
+<TabItem value="app-directory" label="App directory">
+```
 
-If you are using [Hanko Cloud](https://cloud.hanko.io), you can find the API URL on your project dashboard.
-If you are self-hosting you need to provide the URL of your running Hanko backend.
+```jsx title="components/HankoAuth.tsx" showLineNumbers
+"use client";
 
-:::
+import { useEffect } from "react";
+import { register } from "@teamhanko/hanko-elements";
+
+const hankoApi = process.env.HANKO_API_URL;
 
-```jsx title="HankoAuth.jsx" showLineNumbers
+export default function HankoAuth() {
+  useEffect(() => {
+    register(hankoApi).catch((error) => {
+      // handle error
+    });
+  }, []);
+
+  return <hanko-auth />;
+}
+```
+
+```mdx-code-block
+</TabItem>
+<TabItem value="pages-directory" label="Pages directory">
+```
+
+```jsx title="components/HankoAuth.tsx" showLineNumbers
+import { useEffect } from "react";
 import { register } from "@teamhanko/hanko-elements";
 
-const hankoApi = "<YOUR_API_URL>";
+const hankoApi = process.env.HANKO_API_URL;
 
 export default function HankoAuth() {
   useEffect(() => {
-    // register the component
-    // see: https://github.com/teamhanko/hanko/blob/main/frontend/elements/README.md#script
-    register(hankoApi)
-      .catch((error) => {
-        // handle error
-      });
+    register(hankoApi).catch((error) => {
+      // handle error
+    });
   }, []);
```

Comment:
Same but.. we don't have server side components with pages directory so this also would raise errors I feel. Can you check this as well.

Reply:
Hey adding NEXT_PUBLIC prefix, will work here too. thanks for pointing out